### PR TITLE
docket: Add docket agent, split into grid desk

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Adds a docket agent that manages grid tiles. It handles:
- Serving the files && fetching globs (deprecating file-server & glob agents)
- Discovery of package metadata
- Removal of packages

To test:
- boot a galaxy and a star under that galaxy with the pill in this branch
- `|merge %base-dev our %base, =gem %init`
- `|mount %base-dev`
- `rm -rf zod/base-dev/*`
- `rsync -zr src/pkg/base-dev/* zod/base-dev`
- `|commit %base-dev`
- `|merge %grid our %base-dev`
- `|mount %grid`
- `rsync -zr src/pkg/grid/* zod/grid`
- `|commit %grid`
- `|public %grid`
On the star:
- `|install ~zod %grid`
Once install finishes, you should be able to load the star's url in your browser and see the grid interface

testing landscape:
- on the galaxy:
- `|merge %landscape our %base-dev`
- `|mount %landscape`
- `rsync -zr src/pkg/landscape/* zod/landscape`
- `|commit %landscape`
- `|install ~zod %landscape`
- load the star, type `~zod/landscape` in the omnibox and hit enter, it should give you a rendering of the packages metadata, from which you can click 'Get App' and it will install the app.

Notes:
- Does not currently work with devserver. I have the devserver working but `ted/dist` needs to merge master before I add those patches
- Should the `sur/` files for %hood be moved to base-dev?

Demo: 

https://user-images.githubusercontent.com/46801558/128978439-efed4285-3752-4ff3-a25d-4c864aef88d1.mov

### CODE REVIEW

File: pkg/grid/ted/make-glob.hoon [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+=,  strand=strand:spider
+^-  thread:spider
+|=  arg=vase
+=/  m  (strand ,vase)
+^-  form:m
+=+  !<([~ [=desk dir=path]] arg)
+;<  =bowl:spider  bind:m  get-bowl:strandio
+=/  home=path  /(scot %p our.bowl)/[desk]/(scot %da now.bowl)
+=+  .^(paths=(list path) %ct (weld home dir))
+=/  =glob:docket
```
Comment (ryjm) [inline pkg/grid/ted/make-glob.hoon:12]: ah, this is how you make globs! i'll keep my other comment just to note that we should document this.

**Review by ryjm — COMMENTED** [review]
I mostly want more explicit documentation on stuff. Maybe I can help with this while I port srrs. 

Also think there should be more affordances for devs writing new apps (really don't like calling them apps btw, not that i have a better idea) - but that can be done in a separate PR.

File: pkg/grid/ted/make-glob.hoon (hunk 1 above)
Comment (ryjm) [reply pkg/grid/ted/make-glob.hoon:12]: ah i see your other PR has some stuff about this. still think there should be something here that gives you that info though.

**Comment by ryjm** [comment]
also fyi, hark-graph-hook has an unresolved merge conflict in this branch

**Comment by ryjm** [comment]
i was able to install grid from a galaxy to a star, but not landscape for some reason. tells me i'm not running %docket yet. do i install grid on the galaxy as well? when i tried that i get a docket sync error after trying to install %landscape.

```
docket-sync: couldn't subscribe to foreign
watch-ack
/app/docket/hoon:<[395 18].[395 42]>
/app/docket/hoon:<[395 5].[398 51]>
/app/docket/hoon:<[367 27].[367 42]>
/app/docket/hoon:<[367 5].[371 78]>
/app/docket/hoon:<[366 5].[371 78]>
/app/docket/hoon:<[94 15].[94 55]>
/app/docket/hoon:<[94 7].[94 55]>
/app/docket/hoon:<[93 7].[94 55]>
/app/docket/hoon:<[92 7].[94 55]>
/app/docket/hoon:<[91 7].[94 55]>
/app/docket/hoon:<[85 5].[99 7]>
/app/docket/hoon:<[84 3].[100 15]>
/app/docket/hoon:<[83 3].[100 15]>
/sys/vane/gall/hoon:<[1.338 9].[1.338 37```

**Comment by liam-fitzgerald** [comment]
superseded by #5176

**Pull request closed without merging**